app/src/main/java/com/github/libretube/receivers/NotificationReceiver.kt

```diff
@@ -0,0 +1,33 @@
+package com.github.libretube.receivers
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import com.github.libretube.services.DownloadService
+
+class NotificationReceiver : BroadcastReceiver() {
+    override fun onReceive(context: Context?, intent: Intent?) {
+        if (intent?.action == null) return
+
+        val serviceIntent = Intent(context, DownloadService::class.java)
+        serviceIntent.action = intent.action
+
+        val id = intent.getIntExtra("id", -1)
+        if (id == -1) return
+        serviceIntent.putExtra("id", id)
+
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.O) {
```

Comment:
Since we use that multiple times within the whole code we could write a function/compat for that (starting foreground services if supported).